java/core/src/main/java/sleeper/core/range/RegionCanonicaliser.java

```diff
@@ -42,6 +42,9 @@ public static Region canonicaliseRegion(Region region) {
     }
 
     public static boolean isRegionInCanonicalForm(Region region) {
+        if (region == null) {
+            return true;
+        }
```

Comment:
Why is it accepting a null here?

Reply:
Region is allowed to be null, and in multiple places we run region as null (i believe in tests) to test if it works with null

Reply:
It looks like the Partition class was already allowing null in the region field, although this should only ever be the case in tests. There are tests that rely on that, although hopefully we can stop them doing that and maybe add validation in the future. The follow-up issues from this will go a long way towards that.